packages/siwe/lib/client.ts

```diff
@@ -189,165 +171,98 @@ export class SiweMessage {
-      try {
-        addr = utils.verifyMessage(EIP4361Message, signature);
-      } catch (e) {
-        console.error(e);
+      if (!isValidDateObj(notBefore)) {
+        throw new Error(`${notBefore} is not a valid date object`);
       }
-      /** Match signature with message's address */
-      if (addr === this.address) {
-        return resolve({
-          success: true,
-          data: this,
-        });
-      } else {
-        const EIP1271Promise = checkContractWalletSignature(this, signature, opts.provider)
-          .then(isValid => {
-            if (!isValid) {
-              return {
-                success: false,
-                data: this,
-                error: new SiweError(
-                  SiweErrorType.INVALID_SIGNATURE,
-                  addr,
-                  `Resolved address to be ${this.address}`
-                ),
-              };
-            }
-            return {
-              success: true,
-              data: this,
-            };
-          })
-          .catch(error => {
-            return {
-              success: false,
-              data: this,
-              error,
-            };
-          });
-
-        Promise.all([
-          EIP1271Promise,
-          opts?.verificationFallback?.(params, opts, this, EIP1271Promise)?.then(res => res)?.catch((res: SiweResponse) => res)
-        ]).then(([EIP1271Response, fallbackResponse]) => {
-          if (fallbackResponse) {
-            if (fallbackResponse.success) {
-              return resolve(fallbackResponse);
-            } else {
-              fail(fallbackResponse);
-            }
-          } else {
-            if (EIP1271Response.success) {
-              return resolve(EIP1271Response);
-            }
-            else {
-              fail(EIP1271Response);
-            }
-          }
-        });
+
+      if (checkTime.getTime() < notBefore.getTime()) {
+        throw new SiweError(
+          SiweErrorType.NOT_YET_VALID_MESSAGE,
+          `${checkTime.toISOString()} >= ${notBefore.toISOString()}`,
+          `${checkTime.toISOString()} < ${notBefore.toISOString()}`
+        );
       }
-    });
+    }
+
+    let EIP4361Message = this.prepareMessage();
+    
+    /** Normalize signature before verifying */
+    const normalizedSignatureBuf = Buffer.alloc(65); // (drop trailing bytes after recovery byte)
+    normalizedSignatureBuf.write(signature.substring(2), 'hex'); 
+    const normalizedSignature = '0x' + normalizedSignatureBuf.toString('hex');
+
+    /** Recover address from signature */
+    let addr = utils.verifyMessage(EIP4361Message,  normalizedSignature);
+
+    /** Match signature with message's address */
+    if (addr === this.address) {
+      return;
+    }
+
+    const isValid = await checkContractWalletSignature(this, normalizedSignature, opts.provider);
+
+    if (!isValid) {
+      throw new SiweError(
+        SiweErrorType.INVALID_SIGNATURE,
+        addr,
+        `Resolved address to be ${this.address}`
```

Comment:
Shouldn't it be the other way around with the resolved address being `addr`?

Reply:
I'm not sure, the message isn't very clear but i think this is intended to mean `addr` doesn't match what we expected in `this.address`?

Reply:
Yes I just meant, in the definition of `SiweError, the second parameter is the expected value and then the actual value. But it probably doesn't matter